### PR TITLE
Docs: Fix the overindentation of lists

### PR DESCRIPTION
**Pull request by sphuber**
There were a considerable number of bulleted and enumerated lists whose line markers were indented. This would cause them to be rendered as an indented code block. Instead, the lists should be indented to exactly the same level as the preceding text paragraph.

### CODE REVIEW

File: docs/source/howto/data.rst [hunk 1]
```diff
@@ -101,15 +101,15 @@ Once stored, this data can always be retrieved through the ``source`` property:
...
-    * 'license': A string with the type of license that applies to the data, if any.
+* 'db_name': The name of the external database.
+* 'db_uri': The base URI of the external database.
+* 'uri': The exact URI of where the data can be retrieved. Ideally this is a persistent URI.
+* 'id': The external ID with which the data is identified in the external database.
+* 'version': The version of the data, if any.
+* 'extras': Optional dictionary with other fields for source description.
+* 'source_md5': MD5 checksum of the data.
+* 'description': Human-readable free form description of the data's source.
+* 'license': A string with the type of license that applies to the data, if any.
```
Comment (unkcpz) [inline docs/source/howto/data.rst:112]: Since you are already here. I don't think single quotes are needed here as the name of the keys.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @sphuber, just one minor request. All changes looks good.
I am wondering why we don't use any pre-commit hook for reStructuredText? I find there are some such as https://github.com/peterjc/flake8-rst-docstrings, can we adopt it in aiida-core?

File: docs/source/howto/data.rst (hunk 1 above)
Comment (sphuber) [reply docs/source/howto/data.rst:112]: You want me to change them to backticks to make them inline code, or just remove the quotes entirely? Think backticks would be good, right?

**Comment by sphuber** [comment]
> Thanks @sphuber, just one minor request. All changes looks good. I am wondering why we don't use any pre-commit hook for reStructuredText? I find there are some such as https://github.com/peterjc/flake8-rst-docstrings, can we adopt it in aiida-core?

This would be only for the docstrings of the code though, wouldn't it? It wouldn't lint the docs. Anyway, I would be in favor of this, but maybe just use `pydocstyle` in that case. We already use it for `aiida-quantumespresso` and other plugin packages. Haven't added it to `aiida-core` yet since it would be a lot of work to correct all the docstrings. But I guess we could of course start with a blacklist of all files that need to be fixed and we can progressively fix them.

File: docs/source/howto/data.rst (hunk 1 above)
Comment (unkcpz) [reply docs/source/howto/data.rst:112]: I have no preference, to be honest, I think just removing the quotes is clear enough.

**Comment by unkcpz** [comment]
> This would be only for the docstrings of the code though, wouldn't it? It wouldn't lint the docs. Anyway, I would be in favor of this, but maybe just use pydocstyle in that case.

I see, thanks.  But not related to this PR anyway.

File: docs/source/howto/data.rst (hunk 1 above)
Comment (sphuber) [reply docs/source/howto/data.rst]: Had another look and since they are keys in the database, I think double backticks are more appropriate.

**Pull request merged**